sybil/parsers/codeblock.py

```diff
@@ -32,11 +32,28 @@ class CodeBlockParser(object):
         in each of the examples found by this parser.
     """
 
+    _LANGUAGES = {
+        'python': evaluate_code_block,
+    }
+
+    @classmethod
+    def add_evaluator(cls, language, eval_function):
+        """
+        Adds (or overwrites) an evaluator function for the language.
+        """
+        cls._LANGUAGES[language] = eval_function
+
     def __init__(self, future_imports=()):
```

Comment:
How about having `evaluator` as a parameter here, that defaults to `evaluate_code_block`, which could also be renamed to `python_evaluator`?

Reply:
Then how would the dispatch to different `CodeBlockParser` instances work?

Reply:
You're right, I forgot about the ReST language marker too. How about:

```python
pytest_collect_file = Sybil(
    parsers=[
        CodeBlockParser(language='cython', evaluator=cython_evaluator),
        CodeBlockParser(future_imports=['print_function']),
    ],
    pattern='*.rst',
).pytest()
```

...and of course, we should then also support:

```python

class CythonCodeBlockParser(CodeBlockParser):

    language='cython'

    def evaluator(...):
        ...
```